src/com/degel/re_frame_firebase/database.cljs

```diff
@@ -55,24 +55,27 @@
         transaction->js)))
 
 (defn- transactioner [{:keys [path transaction-update on-success on-failure apply-locally]}]
-  (.transaction (fb-ref path)
-                (transaction-update-wrapper transaction-update)
-                (success-failure-wrapper on-success on-failure)
-                ;; Force apply-locally to be a boolean, as required by .transaction.
-                (if (or (false? apply-locally)
-                        (nil? apply-locally))
-                  false
-                  true)))
+  (try
+    (.transaction (fb-ref path)
+                  (transaction-update-wrapper transaction-update)
+                  (success-failure-wrapper on-success on-failure)
+                  ;; Force apply-locally to be a boolean, as required by .transaction.
+                  (if (or (false? apply-locally)
+                          (nil? apply-locally))
+                    false
+                    true))
+    (catch :default e (on-failure e))))
+
 
 (def transaction-effect transactioner)
 
 (defn- swapper [{:keys [path f argv on-success on-failure apply-locally]}]
   (transactioner
```

Comment:
@nodename 
Could you add a docstring on this function too?

Reply:
I'm not going to add docstrings. I know nothing aobut these functions other than what you can understand from their names.

Reply:
alright then